event-schema/counter_manager.go

```diff
@@ -14,14 +14,32 @@ func init() {
 	}
 }
 
-func populateFrequencyCounters(schemaHash string, frequencyCounters []*FrequencyCounter) {
+// populateFrequencyCountersBounded is responsible for capturing the frequency counters which
+// are available in the db and store them in memory but in a bounded manner.
+func populateFrequencyCountersBounded(schemaHash string, frequencyCounters []*FrequencyCounter, bound int) {
 	frequencyCountersMap := make(map[string]*FrequencyCounter)
+	count := 0
```

Comment:
You don't really need `count` here, you could just use the slice index.

As you can see from [this example](https://go.dev/play/p/oaO92Vcnlqy) I created, no matter how you "slice it", when ranging the index always starts from zero. Meaning you could use `index` instead of count.